### ISSUE TITLE
False positive meta-charset-utf-8 error

### ISSUE THREAD

**Issue by paulschreiber** [sonarwhal/sonarwhal#789]
We are seeing a `meta-charset-utf-8` error on our site. It _is_ the first thing in the `<head>`.

Report:
https://sonarwhal.com/scanner/20a37912-3eed-46de-a03a-1bf005c9d2f6

Error:
```
Charset meta tag should be the first thing in '<head>'
https://theundefeated.com/:2:5
<meta charset="UTF-8">
```

Page source:
```
<!DOCTYPE html>
<html lang="en" class="no-js">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="profile" href="http://gmpg.org/xfn/11">
	<link rel="pingback" href="https://theundefeated.com/xmlrpc.php">
```

**Comment by molant** [comment]
The problem with that site is that there's a bunch of scripts added before that tag (I guess on load?):

![image](https://user-images.githubusercontent.com/606594/35582012-5683c4b6-05a2-11e8-955f-3202dd0c1c7c.png)

`sonarwhal` analyzes the site after a few seconds and that's why this rule fails. 
This is an interesting issue. Thanks for reporting it.

**Comment by molant** [comment]
To fix this, the rule should subscribe to the `targetfetch::end` so we get the initial markup.
From there we have 2 options:
1. Use a regexp to check it's on the top
1. Use a package like [`cheerio`](https://www.npmjs.com/package/cheerio) to parse the markup and check it's the first one.

I think I prefer the `cheerio` options just because the web is a mess and we don't know what we might find that breaks our regexp.

**Comment by molant** [comment]
@paulschreiber, I've opened https://github.com/sonarwhal/sonarwhal/pull/900 to address this.
